src/ngAnimate/animateQueue.js

```diff
@@ -624,25 +638,29 @@ var $$AnimateQueueProvider = ['$animateProvider', function($animateProvider) {
         // there is no need to continue traversing at this point
         if (parentAnimationDetected && animateChildren === false) break;
 
-        if (!rootElementDetected) {
-          // angular doesn't want to attempt to animate elements outside of the application
-          // therefore we need to ensure that the rootElement is an ancestor of the current element
-          rootElementDetected = isMatchingElement(parentElement, $rootElement);
-          if (!rootElementDetected) {
-            parentHost = parentElement.data(NG_ANIMATE_PIN_DATA);
-            if (parentHost) {
-              parentElement = parentHost;
-              rootElementDetected = true;
-            }
+        if (!rootElementDetected && !parentHost) {
```

Comment:
This change assumes that there can only be one "pinning" on the path from any element to the rootElement.
E.g. one can't pin element A to element B and element B to element C (where only C is a child of the rootElement).

Although this was not explicitly supported (nor do I know why one might want to do it), it is a potential BC.